pages/model/[uuid].tsx

```diff
@@ -108,12 +111,47 @@ function Model() {
     setCopyModelCardSnackbarOpen(false)
   }
 
+  useEffect(() => {
+    if (currentUser) {
+      if (currentUser !== undefined && currentUser.id === version?.metadata.contacts.manager) {
+        updateLastViewed('manager')
+      }
+      if (currentUser !== undefined && currentUser.id === version?.metadata.contacts.reviewer) {
+        updateLastViewed('reviewer')
+      }
+    }
+  }, [managerLastViewed, reviewerLastViewed, currentUser])
+
+  const updateLastViewed = (role: string) => {
```

Comment:
We should steer clear of type casting and null-null assertions as much as we can. They both nullify the type-safety that TS gives us. In line with https://github.com/gchq/Bailo/pull/149/files?diff=unified&w=1#r982569243 I recommend changing this to:
```
  const updateLastViewed = useCallback(
    (role: string) => {
      if (version && currentUser?.id === version.metadata.contacts.manager) {
        const updatedAtDate = new Date(version.updatedAt)
        if (
          (managerLastViewed && managerLastViewed.getTime() < updatedAtDate.getTime()) ||
          (reviewerLastViewed && reviewerLastViewed.getTime() < updatedAtDate.getTime())
        ) {
          setShowLastViewedWarning(true)
        }
        putEndpoint(`/api/v1/version/${version._id}/lastViewed/${role}`)
      }
    },
    [currentUser, version, managerLastViewed, reviewerLastViewed]
  )
```

Reply:
I've realised after spinning up the app that `managerLastViewed` and `reviewerLastViewed` are ISO strings, so the second `if` statement should instead be:
```
        if (
          (managerLastViewed && new Date(managerLastViewed).getTime() < updatedAtDate.getTime()) ||
          (reviewerLastViewed && new Date(reviewerLastViewed).getTime() < updatedAtDate.getTime())
        )
```